pkg/resource/podtemplatespec.go

```diff
@@ -16,22 +16,29 @@ import (
 
 	configapiv1 "github.com/openshift/api/config/v1"
 	v1 "github.com/openshift/api/imageregistry/v1"
+	operatorapiv1 "github.com/openshift/api/operator/v1"
 	configlisters "github.com/openshift/client-go/config/listers/config/v1"
 
 	"github.com/openshift/cluster-image-registry-operator/pkg/defaults"
 	"github.com/openshift/cluster-image-registry-operator/pkg/storage"
 )
 
+// generateLogLevel returns the appropriate operand log level according to user
+// provided configuration.
 func generateLogLevel(cr *v1.Config) string {
-	switch cr.Spec.LogLevel {
-	case 0:
-		return "error"
+	switch cr.Spec.Logging {
 	case 1:
 		return "warn"
 	case 2, 3:
 		return "info"
 	}
-	return "debug"
+
+	switch cr.Spec.LogLevel {
+	case operatorapiv1.Debug, operatorapiv1.Trace, operatorapiv1.TraceAll:
+		return "debug"
+	default:
+		return "info"
```

Comment:
Changed this, PTAL. Basically, if Loglevel is set to operatorapiv1.Debug, operatorapiv1.Trace or operatorapiv1.TraceAll we set it to debug, otherwise (operatorapiv1.Normal) we fall back to Logging (but this time if it is zero we return "info" instead of "error" as before)